### PR TITLE
Lossless 598: Fixed varios issues

### PR DESCRIPTION
**Pull request by Ignacio-Freire**

### CODE REVIEW

**Comment by DomantasP** [comment]
setStakingToken tests are still missing test for zero address

**Comment by DomantasP** [comment]
LOSSLESS-608. Fee should be renamed reward in solidity code too.

File: test/contract based tests/EnvironmentTests.js [hunk 1]
```diff
@@ -27,6 +28,40 @@ describe('Lossless Environment', () => {
...
+        ).to.not.be.reverted;
+        await expect(
+          env.lssReporting.connect(adr.lssAdmin).setCommitteeReward(1),
+        ).to.not.be.reverted;
+        await expect(
+          env.lssReporting.connect(adr.lssAdmin).setReporterReward(4),
+        ).to.not.be.reverted;
+
+        expect(
+          await env.lssReporting.getRewards(),
```
Comment (DomantasP) [inline test/contract based tests/EnvironmentTests.js:62]: I would recommend verifying that returned values are correct.

File: contracts/LosslessReporting.sol [hunk 2]
```diff
@@ -208,7 +211,7 @@ contract LosslessReporting is Initializable, ContextUpgradeable, PausableUpgrade
     /// @param account Potential malicious address
     function secondReport(uint256 reportId, address account) public notBlacklisted whenNotPaused {
         require(account != address(0), "LSS: Cannot report zero address");
-        require(!losslessGovernance.isReportSolved(reportId), "LSS: Report already solved.");
+        require(!losslessGovernance.isReportSolved(reportId) && !losslessGovernance.reportResolution(reportId), "LSS: Report already solved.");
```
Comment (DomantasP) [inline contracts/LosslessReporting.sol:211]: Did this not require any updates in the tests?

File: contracts/LosslessReporting.sol [hunk 3]
```diff
@@ -177,7 +177,10 @@ contract LosslessReporting is Initializable, ContextUpgradeable, PausableUpgrade
 
         uint256 reportId = tokenReports[token].reports[account];
 
-        require(reportId == 0 || reportTimestamps[reportId] + reportLifetime < block.timestamp || losslessGovernance.isReportSolved(reportId), "LSS: Report already exists");
+        require(reportId == 0 || 
+                reportTimestamps[reportId] + reportLifetime < block.timestamp || 
+                losslessGovernance.isReportSolved(reportId) && 
+                !losslessGovernance.reportResolution(reportId), "LSS: Report already exists");
```
Comment (DomantasP) [inline contracts/LosslessReporting.sol:180]: Did this not require any updates in the tests?

File: contracts/LosslessReporting.sol (hunk 2 above)
Comment (DomantasP) [reply contracts/LosslessReporting.sol:211]: Cannot we have the same validation statement as in the _report_ function ? 

`require(reportId == 0 || 
                reportTimestamps[reportId] + reportLifetime < block.timestamp || 
                losslessGovernance.isReportSolved(reportId) && 
                !losslessGovernance.reportResolution(reportId), "LSS: Report already exists");
`
?

File: contracts/LosslessReporting.sol (hunk 3 above)
Comment (Ignacio-Freire) [reply contracts/LosslessReporting.sol:180]: I have been testing this part quite a bit and it's actually not doing much as all this is assigned by the code. This part only makes sense to have on `secondReport`.

`reportId == 0` checks that it cannot be a report number zero. This is never the case as the number of report gets assigned here.

`reportTimestamps[reportId] + reportLifetime < block.timestamp` Chechks that the report is not expired. This can never happen as the report is being created at this time. 

`losslessGovernance.isReportSolved(reportId)` Again, the `reportId `is assigned on this same method and the `reportTimestamp` gets assigned after.

`!losslessGovernance.reportResolution(reportId)` Same as before, the `reportId` is being assigned on this method. 

I've been trying to force some tests in order to generate a report at the exact same time with the same Id but it's not possible. Maybe we can test this from the front end. 

One reason to leave this checks is if some error happens where two reports are getting the same Id. This could happen if someone find a way to exploit this particular method. 

Otherwise it doesn't make much sense to have it on `report()`

**Comment by Ignacio-Freire** [comment]
> 

Fixed on

File: test/contract based tests/EnvironmentTests.js (hunk 1 above)
Comment (Ignacio-Freire) [reply test/contract based tests/EnvironmentTests.js]: Added on ee22d4a

**Review by DomantasP — APPROVED** [review]
